cypress/integration/buy-shirt.cy.ts

```diff
@@ -1,28 +1,34 @@
+import {AddressStepPage, LoginPage, MenuContentPage, PaymentStepPage,
+  ProductsListPage, ShippingStepPage, ShoppingCartPage} from "../page/index";
+
+const menuContentPage = new MenuContentPage();
+const productLists = new ProductsListPage();
+const shoppingCartPage = new ShoppingCartPage();
+const loginPage = new LoginPage();
+const addressStepPage = new AddressStepPage();
+const shippingStepPage = new ShippingStepPage();
+const paymentStepPage = new PaymentStepPage();
+
 describe("Buy a t-shirt", () => {
-  it("then the t-shirt should be bought", () => {
-    cy.visit("http://automationpractice.com/");
-    cy.get("#block_top_menu > ul > li:nth-child(3) > a").click();
-    cy.get(
-        "#center_column a.button.ajax_add_to_cart_button.btn.btn-default",
-    ).click();
-    cy.get("[style*='display: block;'] .button-container > a").click();
-    cy.get(".cart_navigation span").click();
-
-    cy.get("#email").type("aperdomobo@gmail.com");
-    cy.get("#passwd").type("WorkshopProtractor");
-
-    cy.get("#SubmitLogin").click();
-    cy.get("[name=\"processAddress\"]").click({force: true});
-
-    cy.get("#cgv").check();
-    cy.get("[name=\"processCarrier\"]").click();
-
-    cy.get(".bankwire").click();
-    cy.get("#cart_navigation").find("button").click();
-
-    cy.get("#center_column > div > p > strong").should(
-        "have.text",
-        "Your order on My Store is complete.",
-    );
+  it("then should be bought a t-shirt", () => {
+    menuContentPage.visitMenuContentPage();
+    menuContentPage.goToTShirtMenu();
+
+    productLists.addShirtToCart();
+    productLists.clickProceedToCheckout();
+
+    shoppingCartPage.clickProceedToCheckout();
+
+    loginPage.login();
+
+    addressStepPage.clickProceedToCheckout();
+
+    shippingStepPage.checkTermsBox();
+    shippingStepPage.clickProceedToCheckout();
```

Comment:
There is no need in mentioning that is a click, you can just rename it as proceedToCheckout